### PR TITLE
Support signing messages for external accounts

### PR DESCRIPTION
**Pull request by Tbaut**
I implemented the signing support in Stylo and submitted a PR for Signer.
It's working well with apps and Polkassembly.

https://user-images.githubusercontent.com/33178835/118889678-bce08700-b8fd-11eb-9e71-5fbf285d8082.mp4

### CODE REVIEW

**Comment by jacogr** [comment]
Sorry for the delay, will try to merge today - just one final look.

**Comment by Tbaut** [comment]
No problem at all. I've added the genesis hash in the QR code here. We had a discussion with @Slesarew and although the genesis is not absolutely mandatory (after all the genesis shouldn't change anything in the signature), because the accounts are generally identified with a public key + genesis, on Signer/Stylo as well as the extension, it would make sense to keep it in message signing as well.

**Comment by Slesarew** [comment]
> although the genesis is not absolutely mandatory 
Well, here, I must admit, it IS mandatory here after all. If we start signing arbitrary messages, that will be another uos call. The question is whether we always need another genhash in transactions, but it's out of scope here.

File: packages/extension-ui/src/Popup/Signing/Qr.tsx [hunk 1]
```diff
@@ -2,50 +2,69 @@
...
     []
   );
 
+  if (!payloadU8a) {
+    return (<div className={className}>
+      <div className='qrContainer'>
+        Transaction command:{cmd} not supported.
+      </div>
+    </div>);
+  }
```
Comment (jacogr) [inline packages/extension-ui/src/Popup/Signing/Qr.tsx:56]:
```suggestion
  if (!payloadU8a) {
    return (
      <div className={className}>
        <div className='qrContainer'>
          Transaction command:{cmd} not supported.
        </div>
      </div>
    );
  }
```

File: packages/extension-ui/src/Popup/Signing/Qr.tsx [hunk 2]
```diff
@@ -2,50 +2,69 @@
...
 
   const _onShowQr = useCallback(
     () => setIsScanning(true),
     []
   );
 
+  if (!payloadU8a) {
+    return (<div className={className}>
+      <div className='qrContainer'>
+        Transaction command:{cmd} not supported.
```
Comment (jacogr) [inline packages/extension-ui/src/Popup/Signing/Qr.tsx:53]: This should really be wrapped in `t('...')`, but I guess it is not absolutely critical ... if we do get here we probably have bigger issues...

File: packages/extension-ui/src/Popup/Signing/Request/index.tsx [hunk 3]
```diff
@@ -123,20 +130,32 @@ export default function Request ({ account: { accountIndex, addressOffset, isExt
...
+              cmd={CMD_SIGN_MESSAGE}
+              genesisHash={account.genesisHash}
+              onSignature={_onSignature}
+              payload={data}
+            />
+          )
+          : <Bytes
+            bytes={data}
+            url={url}
+          />
```
Comment (jacogr) [inline packages/extension-ui/src/Popup/Signing/Request/index.tsx:157]:
```suggestion
          : (
            <Bytes
              bytes={data}
              url={url}
            />
          )
```

**Comment by jacogr** [comment]
Some OCD-like comments, but seems sane.

File: packages/extension-ui/src/Popup/Signing/Qr.tsx (hunk 2 above)
Comment (Tbaut) [reply packages/extension-ui/src/Popup/Signing/Qr.tsx:53]: Thanks, totally, this slipped my mind. Will add it.

**Review by jacogr — APPROVED** [review]
As always, I really do appreciate your efforts. 

Exceptionally happy to see this in and supported, it has been a massive PITA with requests and support overhead.

**Pull request merged**